### DISCUSSION TITLE
Keyframe Tangent Data in USD

### DISCUSSION THREAD

**Discussion by MrLockness** [Autodesk/maya-usd#1812] [Q&A]
We're using USD for saving unbaked animation for a proprietary animation library (inside of Maya). So far I have created custom attributes to handle the keyframes tangent data, but the client have asked to make sure that there is no already existing support for animated attributes in USD API. 

From my knowledge animated attribute only support time stamp and value. My question is, is this correct? I didn't miss anything in the documentation?

My second question, is there any plan to extend the parameters saved to animated attributes beyond the value and timestamp.

I know this is a very specific case use but I just want to be able to give the client a clear and correct answer.

Thanks for taking the time answering!

**Comment by kxl-adsk** [comment]
> From my knowledge animated attribute only support time stamp and value. My question is, is this correct? I didn't miss anything in the documentation?

That is correct.

> My second question, is there any plan to extend the parameters saved to animated attributes beyond the value and timestamp.

This is a problem we will have to tackle together at the USD community level. If you haven't seen it, there is a great document authored by @spiffmon "On Rigging And Execution in USD" which touched on the subject of Animation Curves.

**Comment by spiffmon** [comment]
Yes, thanks, @kxl-adsk , Animation Curves _are_ on our radar, but only a distant blip behind many other projects right now.  Judging by the number of strong opinions that were registered during the "opening of AnimX" discussions several years ago, it will be fun to come together, especially as Pixar will want to try to retain close compatibility with Presto animation curves, which do have some unique features not broadly supported in other DCC's.
